shell/platform/common/cpp/public/flutter_texture_registrar.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_PUBLIC_FLUTTER_TEXTURE_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_PUBLIC_FLUTTER_TEXTURE_REGISTRAR_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "flutter_export.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+struct FlutterDesktopTextureRegistrar;
+// Opaque reference to a texture registrar.
+typedef struct FlutterDesktopTextureRegistrar*
+    FlutterDesktopTextureRegistrarRef;
+
+typedef enum {
+  // A Pixel buffer-based texture.
+  kFlutterDesktopPixelBufferTexture
+} FlutterDesktopTextureType;
+
+// An image buffer object.
+typedef struct {
+  // The pixel data buffer.
+  const uint8_t* buffer;
+  // Width of the pixel buffer.
+  size_t width;
+  // Height of the pixel buffer.
+  size_t height;
+} FlutterDesktopPixelBuffer;
+
+// The pixel buffer copy callback definition provided to
+// the Flutter engine to copy the texture.
+typedef const FlutterDesktopPixelBuffer* (
+    *FlutterDesktopPixelBufferTextureCallback)(size_t width,
+                                               size_t height,
+                                               void* user_data);
+
+// An object used to configure pixel buffer textures.
+typedef struct {
+  // The callback used by the engine to copy the pixel buffer object.
+  FlutterDesktopPixelBufferTextureCallback callback;
+  // Opaque data that will get passed to the provided |callback|.
+  void* user_data;
+} FlutterDesktopPixelBufferTextureConfig;
+
+typedef struct {
+  FlutterDesktopTextureType type;
+  union {
+    FlutterDesktopPixelBufferTextureConfig pixel_buffer;
```

Comment:
Nit: Let's call this `pixel_buffer_config` so that it doesn't sound like a `FlutterDesktopPixelBuffer` instance.